### PR TITLE
[native] Fix tpch connector unit test for presto on spark native

### PR DESCRIPTION
**Pull request by vermapratyush**
Register tpchstandard catalog and fix missing connector expected message.
Allow injecting custom NativeExecutionConnectorConfig to NativeExecutionModule 
to override connector name which otherwise defaults to hive.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeTpchConnectorQueries.java [hunk 1]
```diff
@@ -13,20 +13,29 @@
...
 import com.facebook.presto.testing.QueryRunner;
 
 public class TestPrestoNativeTpchConnectorQueries
         extends AbstractTestNativeTpchConnectorQueries
 {
     @Override
-    protected QueryRunner createQueryRunner() throws Exception
+    public Session getSession()
+    {
+        return Session.builder(super.getSession()).setCatalog("tpchstandard").setSchema("tiny").build();
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeTpchConnectorQueries.java:26]: This code seems the same between Prestissimo and Presto-on-Spark tests. It would be nice to avoid duplicating it.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeTpchConnectorQueries.java [hunk 2]
```diff
@@ -39,8 +40,7 @@ public void testMissingTpchConnector()
     @Test
     public void testTpchTinyTables()
     {
-        Session session = tpchStandardTiny();
-
+        Session session = getSession();
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeTpchConnectorQueries.java:43]: Why this change? If you use 'getSession' than you can call assertQuery without the session parameter.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java [hunk 3]
```diff
@@ -41,16 +43,16 @@ protected void assertQuery(String sql)
...
 
-    // TODO: Enable following Ignored tests after fixing (Tests can be enabled by removing the method)
-    @Override
-    @Ignore
-    public void testMissingTpchConnector() {}
-
-    @Override
-    @Ignore
-    public void testTpchDateFilter() {}
+    @Test(expectedExceptions = PrestoSparkFailure.class, expectedExceptionsMessageRegExp = ".*Connector with ID 'tpch' not registered.*")
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:46]: Use assertQueryFails

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeTpchConnectorQueries.java (hunk 2 above)
Comment (vermapratyush) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeTpchConnectorQueries.java]: Didn't realise assertQuery was overloaded with a better method.
Fixed.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java [hunk 4]
```diff
@@ -41,16 +41,14 @@ protected void assertQuery(String sql)
...
-    public void testMissingTpchConnector() {}
-
-    @Override
-    @Ignore
-    public void testTpchDateFilter() {}
-
-    @Override
-    @Ignore
-    public void testTpchTinyTables() {}
+    public void testMissingTpchConnector()
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:45]: Why can't we use the test from the base class?

File: presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java [hunk 5]
```diff
@@ -339,6 +339,12 @@ public PrestoSparkQueryRunner(
         // Install tpch Plugin
         pluginManager.installPlugin(new TpchPlugin());
         connectorManager.createConnection("tpch", "tpch", ImmutableMap.of());
+        // TPCH-Standard uses tpch column naming
+        // See: https://github.com/prestodb/presto/issues/1771
+        Map<String, String> tpchProperties = ImmutableMap.<String, String>builder()
+                .put("tpch.column-naming", "standard")
+                .build();
+        connectorManager.createConnection("tpchstandard", "tpch", tpchProperties);
```
Comment (mbasmanova) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:347]: I wonder if this is the only change that's really necessary.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java (hunk 4 above)
Comment (vermapratyush) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:45]: The error message is different for presto native and presto on spark native. I have moved the code around a bit to avoid the duplicate.

File: presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java (hunk 5 above)
Comment (vermapratyush) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:347]: Actually, this works on detached mode. For non-detached mode, it requires some more change. Working on it.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java (hunk 4 above)
Comment (mbasmanova) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:44]: Any particular reason for the difference? Should we make it the same?

Comment (vermapratyush) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:44]: Typical spark deployment (including fb-spark) does not support querying multiple catalogue under the same spark session (unless it is a dataset app). So the expected error message is accurate for both native as well as pos native.

Comment (mbasmanova) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:44]: I'd think this limitation would not apply to PoS since it uses PrestoSQL for language, query planning and execution and uses Spark only for distributed execution. Why would that piece care about whether there are multiple catalogs or just one?

Comment (mbasmanova) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:44]: I see. This is quite unfortunate. Let's add a comment to explain this and let's also revisit this design decision.

File: presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java (hunk 5 above)
Comment (vermapratyush) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:337]: This was not working in non-detached mode as the connectorManager has hardcoded connector names. I have refactored how NativeExecutionConnectionManager is injected.

Comment (vermapratyush) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:337]: one more change pending, i will close this PR and re-open when it is pushed.

**Comment by vermapratyush** [comment]
@mbasmanova  There was an issue in detached mode, I have fixed it with some refactoring to how catalog config is injected.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java (hunk 4 above)
Comment (miaoever) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:44]: Does that means we're configuring the C++ worker with different connector name (tpch) from the connector name using on java side (hive)? 
https://github.com/prestodb/presto/blob/master/presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java#L376

Comment (vermapratyush) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:44]: On java side we also configured with tpch connector name (client side we can have multiple connection names and most likely that is how it all worked) https://github.com/prestodb/presto/blob/master/presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java#L341

File: presto-native-execution/src/test/java/com/facebook/presto/spark/PrestoSparkNativeQueryRunnerUtils.java [hunk 6]
```diff
@@ -103,16 +104,29 @@ public static Map<String, String> getNativeExecutionSessionConfigs()
...
-        return queryRunner;
+        return createPrestoSparkNativeQueryRunner(
+                Optional.of(getBaseDataPath()),
+                getNativeExecutionSessionConfigs(),
+                getNativeExecutionShuffleConfigs(),
+                ImmutableList.of(nativeExecutionModule));
+    }
+
+    // Similar to createPrestoSparkNativeQueryRunner, but with custom connector config and without jsonFunctionNamespaceManager
+    public static PrestoSparkQueryRunner createPrestoSparkNativeTPCHQueryRunner()
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/PrestoSparkNativeQueryRunnerUtils.java:125]: These names are long and hard to read. Since PrestoSparkNativeQueryRunner is part of the class name, these methods can be shortened to:

createHiveRunner and createTpchRunner

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java [hunk 7]
```diff
@@ -16,15 +16,14 @@
...
-import org.testng.annotations.Ignore;
 
 public class TestPrestoSparkNativeTpchConnectorQueries
         extends AbstractTestNativeTpchConnectorQueries
 {
     @Override
     protected QueryRunner createQueryRunner()
     {
-        return PrestoSparkNativeQueryRunnerUtils.createPrestoSparkNativeQueryRunner();
+        return PrestoSparkNativeQueryRunnerUtils.createTPCHRunner();
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeTpchConnectorQueries.java:26]: all caps are hard to read; consider createTpchRunner

File: presto-native-execution/src/test/java/com/facebook/presto/spark/PrestoSparkNativeQueryRunnerUtils.java [hunk 8]
```diff
@@ -101,21 +102,34 @@ public static Map<String, String> getNativeExecutionSessionConfigs()
...
+                getNativeExecutionShuffleConfigs(),
+                ImmutableList.of(nativeExecutionModule));
+    }
+
+    // Similar to createPrestoSparkNativeQueryRunner, but with custom connector config and without jsonFunctionNamespaceManager
+    public static PrestoSparkQueryRunner createTPCHRunner()
+    {
+        return createHiveRunner(
+                new NativeExecutionModule(
+                        Optional.of(new NativeExecutionConnectorConfig().setConnectorName("tpch"))));
```
Comment (vermapratyush) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/PrestoSparkNativeQueryRunnerUtils.java:129]: @mbasmanova this override is key to making the tests run.

**Review by mbasmanova — APPROVED** [review]
@vermapratyush Thank you for the fix.
